plugins/pur/core/src/main/java/org/pentaho/di/repository/pur/PurRepository.java

```diff
@@ -380,16 +379,6 @@ public RootRef getRootRef() {
   @Override public void disconnect() {
     connected = false;
     metaStore = null;
-    IMetaStore activeMetaStore = null;
```

Comment:
If we are connected to repository metastore, then disconnecting from repository should trigger a call to go back to MetastoreLocator so it can find which metastore to connect

Reply:
hrm. The next time the Supplier is called, it should return a non-repository metastore, since the repository isn't connected anymore. 
But, if anything is holding on to a metastore instance instead of the supplier, that wouldn't switch over like the DelegatingMetastore did. hrm. I'll dig into this.

Reply:
See updated PR